app/jobs/course/assessment/answer/auto_grading_job.rb

```diff
@@ -8,6 +9,16 @@ class Course::Assessment::Answer::AutoGradingJob < ApplicationJob
   # @param [Course::Assessment::Answer::AutoGrading] auto_grading The object to store the grading
   #   results into.
   def perform_tracked(auto_grading)
-    Course::Assessment::Answer::AutoGradingService.grade(auto_grading)
+    instance = nil
+    submission = auto_grading.answer.submission
+
+    Course.unscoped do
+      instance = submission.assessment.course.instance
+    end
+    ActsAsTenant.with_tenant(instance) do
+      Course::Assessment::Answer::AutoGradingService.grade(auto_grading)
+      redirect_to(edit_course_assessment_submission_path(submission.assessment.course,
```

Comment:
Okay, do we have other cases that gradings not caused by submission ? 

But make this a parameter makes it more flexible :+1: